### PR TITLE
Handle multi-line statsd input more like statsd

### PR DESCRIPTION
**Pull request by wyndhblb**
Some statsd emitters put a double \n\n in the output of multi line statsd stats.  This causes the entire message to be treated as failed, when there is an easy fix here.  Same behavior as statsd itself https://github.com/etsy/statsd/blob/master/stats.js line  205 or so.

### CODE REVIEW

File: plugins/statsd/statsd_input.go [hunk 1]
```diff
@@ -125,20 +125,24 @@ func (s *StatsdInput) Stop() {
...
 	}
 
+	if warn != nil || len(warn) > 0 {
+		s.ir.LogError(fmt.Errorf("Warning Part of message could not be parsed: %s", warn))
+	}
+
 	for _, stat := range stats {
 		if !s.statAccum.DropStat(stat) {
-			s.ir.LogError(fmt.Errorf("Undelivered stat: %v", stat))
+			s.ir.LogError(fmt.Errorf("Undelivered stat: %s", stat))
```
Comment (rafrombrc) [inline plugins/statsd/statsd_input.go]: %v will output the stat values, converting to string where necessary, whereas %s will generate an error for non-string attributes of the struct. If we're going to change it at all, using %+v to include the field names is a better choice.

File: plugins/statsd/statsd_input.go [hunk 2]
```diff
@@ -152,15 +156,34 @@ func parseMessage(message []byte) ([]Stat, error) {
 		lines = [][]byte{message}
 	}
 
-	for _, line := range lines {
+	var num_lines int = len(lines)
+	var bad_lines_ct int = 0
```
Comment (rafrombrc) [inline plugins/statsd/statsd_input.go]: We don't really need to keep a separate count of the bad lines, do we? We can just check len(bad_lines) instead.

File: plugins/statsd/statsd_input_test.go [hunk 3]
```diff
@@ -166,7 +166,7 @@ func TestParseMessage(t *testing.T) {
 	}
 
 	for msg, expected := range testData {
-		obtained, err := parseMessage([]byte(msg + "\n"))
+		obtained, _, err := parseMessage([]byte(msg + "\n"))
```
Comment (rafrombrc) [inline plugins/statsd/statsd_input_test.go]: Probably a good idea to capture this value here (and below) and check that it's what we expect.

**Comment by rafrombrc** [comment]
Overall this looks pretty good, thanks for your contribution! In addition to the in-line notes, it would be great if you could add a note to the CHANGES.txt changelog file, as a new feature for the 0.9 release. Cheers!

File: plugins/statsd/statsd_input.go (hunk 2 above)
Comment (wyndhblb) [inline plugins/statsd/statsd_input.go]: not a real "go" expert, but doing "length" in a few languages can be a performance issue.  if there is not penalty in Go, no there is no need to keep a separate count.

File: plugins/statsd/statsd_input.go (hunk 1 above)
Comment (wyndhblb) [inline plugins/statsd/statsd_input.go]: okey-dokey

File: CHANGES.txt [hunk 4]
```diff
@@ -74,7 +74,13 @@ Features
 
 * Added ability for HttpListenInput to capture specified HTTP request headers
   and write them as message fields.
+  
+* Added ability for Statsd input handler to treat a malformed stat line 
+  one at time without skipping all the "good" stats in a multi-line input
+  (more like statsd itself).
+  
 
+>>>>>>> upstream/dev
```
Comment (bbinet) [inline CHANGES.txt]: merge conflict artifact?

**Pull request merged**